dynawo/sources/Solvers/AlgebraicSolvers/DYNSolverKINAlgRestoration.cpp

```diff
@@ -336,7 +336,9 @@ SolverKINAlgRestoration::solve(bool noInitSetup) {
   }
 
   flag = solveCommon();
-  if (flag < 0)
+  if (flag == KIN_STEP_LT_STPTOL || flag == KIN_INITIAL_GUESS_OK)
```

Comment:
Might be better to add a else with a throw in the case another positive number if returned, otherwise if sundials is updated we might get nasty bugs (solver passes but simulation considers it fails)

Reply:
It is normally speaking already handled by the solveCommon that calls analyseFlag that should throw if we get a status that we don't know how to interprete.

Anyway, as Sonar notices, I don't know why I reported this condition here because we don't use the flag anywhere else in this part, except if it is negative to throw which is not the case of the statuses I transform.

To sum up, I will revert my modification here (no need for something special)